src/software/ai/hl/stp/play/penalty_kick_enemy_play_test.cpp

```diff
@@ -2,50 +2,126 @@
 
 #include <gtest/gtest.h>
 
+#include "software/simulated_tests/non_terminating_validation_functions/enemy_never_scores_validation.h"
 #include "software/simulated_tests/simulated_play_test_fixture.h"
+#include "software/simulated_tests/terminating_validation_functions/robot_in_polygon_validation.h"
+#include "software/simulated_tests/terminating_validation_functions/robot_state_validation.h"
 #include "software/simulated_tests/validation/validation_function.h"
 #include "software/test_util/test_util.h"
 #include "software/time/duration.h"
 #include "software/world/world.h"
 
-class PenaltyKickEnemyPlayTest : public SimulatedPlayTestFixture
+class PenaltyKickEnemyPlayTest
+    : public SimulatedPlayTestFixture,
+      public ::testing::WithParamInterface<
+          std::tuple<RefereeCommand, RefereeCommand, std::vector<RobotStateWithId>>>
 {
    protected:
     Field field = Field::createSSLDivisionBField();
 };
 
-TEST_F(PenaltyKickEnemyPlayTest, test_penalty_kick_enemy_play)
+TEST_P(PenaltyKickEnemyPlayTest, test_penalty_kick_enemy_play_setup)
 {
-    BallState ball_state(field.friendlyPenaltyMark(), Vector(0, 0));
-    auto friendly_robots = TestUtil::createStationaryRobotStatesWithId(
-        {Point(-3, 2.5), Point(-3, 1.5), Point(-3, 0.5), Point(-3, -0.5), Point(-3, -1.5),
-         Point(4.6, -3.1)});
+    RefereeCommand current_command  = std::get<0>(GetParam());
+    RefereeCommand previous_command = std::get<1>(GetParam());
+    BallState ball_state(field.enemyPenaltyMark(), Vector(0, 0));
+
+    std::vector<RobotStateWithId> friendly_robots = std::get<2>(GetParam());
+
+    // enemy robots behind the penalty mark
+    auto enemy_robots = TestUtil::createStationaryRobotStatesWithId({
+        Point(field.enemyPenaltyMark().x() + 0.3, 0),
+        Point(field.enemyPenaltyMark().x() + 1, 0),
+        Point(field.enemyPenaltyMark().x() + 1, 4 * ROBOT_MAX_RADIUS_METERS),
+        Point(field.enemyPenaltyMark().x() + 1, 8 * ROBOT_MAX_RADIUS_METERS),
+        Point(field.enemyPenaltyMark().x() + 1, -4 * ROBOT_MAX_RADIUS_METERS),
+        Point(field.enemyPenaltyMark().x() + 1, -8 * ROBOT_MAX_RADIUS_METERS),
+    });
     setFriendlyGoalie(0);
+    setEnemyGoalie(0);
+    setAIPlay(TYPENAME(PenaltyKickEnemyPlay));
+    setRefereeCommand(current_command, previous_command);
+    Polygon behind_ball =
```

Comment:
It isn't clear in the rest of the test that this is a `Polygon`, should name this something clearer like `behind_ball_region`